internal/pkg/telemetry/metric/prometheus/prometheus.go

```diff
@@ -70,7 +70,8 @@ func ServeMetrics(ctx context.Context, serviceName, listenAddr string, logger lo
 	proc.Add(func(shutdown servicectx.ShutdownFn) {
 		logger.InfofCtx(ctx, `HTTP server listening on "%s/%s"`, listenAddr, Endpoint)
 		serverErr := srv.ListenAndServe() // ListenAndServe blocks while the server is running
-		shutdown(context.Background(), serverErr)
+		// intentionally creating new context for the shutdown operation
+		shutdown(context.Background(), serverErr) // nolint: contextcheck
```

Comment:
Reason can be on the same line:
```
// nolint: contextcheck  // intentionally creating new context for the shutdown operation
```

See `You may add a comment explaining ...`
https://golangci-lint.run/usage/false-positives/